packages/next/src/client/components/router-reducer/reducers/navigate-reducer.ts

```diff
@@ -419,7 +419,11 @@ function navigateReducer_PPR(
               seedData,
               head
             )
-            if (task !== null && task.node !== null) {
+            if (
+              !mutable.onlyHashChange &&
```

Comment:
Fixing the bug in this PR indirectly broke a navigation test, because it was falling back to the forked implementation.

With PPR, we shouldn't spawn a dynamic task if only the hash fragment changed, as the server won't have anything new.